### ISSUE TITLE
no YAeHMOP support on Mac ARM for rdkit 2023.05

### ISSUE THREAD

**Issue by juius** [kuelumbus/rdkit-pypi#96]
Version 2023.5 seems to be compiled without YAeHMOP support, 2022.05 has it on same machine.
Error message comes from
```python
from rdkit import Chem
from rdkit.Chem import rdDetermineBonds
mol = Chem.MolFromXYZBlock("2\n\nH 0.0 0.0 0.0\nH 0.74 0.0 0.0\n")
rdDetermineBonds.DetermineConnectivity(mol,useHueckel=True)
```

**Comment by kuelumbus** [comment]
Thanks for reporting this. I can repo this. 

I can see the YAeHMOP build running in the build logs. ` -DRDK_BUILD_YAEHMOP_SUPPORT=ON` is also set. From `Release_2023_09_3`, rdkit throws the error:

```py
In [1]: from rdkit import Chem
   ...: from rdkit.Chem import rdDetermineBonds
   ...: mol = Chem.MolFromXYZBlock("2\n\nH 0.0 0.0 0.0\nH 0.74 0.0 0.0\n")
   ...: rdDetermineBonds.DetermineConnectivity(mol,useHueckel=True)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[1], line 4
      2 from rdkit.Chem import rdDetermineBonds
      3 mol = Chem.MolFromXYZBlock("2\n\nH 0.0 0.0 0.0\nH 0.74 0.0 0.0\n")
----> 4 rdDetermineBonds.DetermineConnectivity(mol,useHueckel=True)

ValueError: The RDKit was not compiled with YAeHMOP support
```

Seems like there was some restructuring of YAeHMOP in 
https://github.com/rdkit/rdkit/compare/Release_2023_09_2...Release_2023_09_3

@greglandrum can you help?

**Comment by greglandrum** [comment]
This is due to a mistake I made when attempting to make the YAeHMOP support optional for DetermineConnectivity.
I'll fix it for the 2024.03.2 release.
Thanks for pointing out the problem @juius

**Comment by kuelumbus** [comment]
This should be fixed in the new release.
